client/dashboard/sites/scan/components/bulk-fix-threats-modal.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+	fixThreatsMutation,
+	fixThreatsStatusQuery,
+	siteScanQuery,
+	siteScanHistoryQuery,
+} from '@automattic/api-queries';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { __experimentalVStack as VStack, Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect, useCallback } from 'react';
+import { ButtonStack } from '../../../components/button-stack';
+import { Text } from '../../../components/text';
+import { ThreatsDetailCard } from './threats-detail-card';
+import type { Threat } from '@automattic/api-core';
+import type { RenderModalProps } from '@wordpress/dataviews';
+
+interface BulkFixThreatsModalProps extends RenderModalProps< Threat > {
+	siteId: number;
+}
+
+export function BulkFixThreatsModal( { items, closeModal, siteId }: BulkFixThreatsModalProps ) {
+	const bulkFixableThreats = items.filter(
+		( item ) => item.fixable?.extras?.is_bulk_fixable !== false
+	);
+	const bulkFixableIds = new Set( bulkFixableThreats.map( ( item ) => item.id ) );
+	const remainingThreats = items.filter( ( item ) => ! bulkFixableIds.has( item.id ) );
+
+	const queryClient = useQueryClient();
+	const bulkFixThreats = useMutation( fixThreatsMutation( siteId ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const [ isBulkFixInProgress, setIsBulkFixInProgress ] = useState( false );
+
+	const { data: bulkFixStatusData } = useQuery( {
+		...fixThreatsStatusQuery( siteId, Array.from( bulkFixableIds ) ),
+		refetchInterval: isBulkFixInProgress ? 2000 : false,
+		enabled: isBulkFixInProgress,
+	} );
+
+	const handleBulkFixed = useCallback( () => {
+		queryClient.invalidateQueries( siteScanQuery( siteId ) );
+		queryClient.invalidateQueries( siteScanHistoryQuery( siteId ) );
+		closeModal?.();
+		createSuccessNotice( __( 'All threats have been successfully fixed!' ), {
+			type: 'snackbar',
+		} );
+	}, [ closeModal, createSuccessNotice, queryClient, siteId ] );
+
+	useEffect( () => {
+		if ( ! bulkFixStatusData?.threats ) {
+			return;
+		}
+
+		if ( isBulkFixInProgress ) {
+			const allFixed = Object.values( bulkFixStatusData.threats ).every(
+				( threat ) => threat?.status === 'fixed'
+			);
```

Comment:
I would suggest to validate `threat.status !== 'in_progress'` instead, since there could be threats with `no_fixed` status too.

Here is a similar logic you could use:
https://github.com/Automattic/wp-calypso/blob/9ff262ee3217bfa8ff3481b2d69b1fc21133c9eb/client/state/data-layer/wpcom/sites/alerts/fix-status.js#L27

Message for when all threats have been fixed:
`All threats were successfully fixed.`

Message for when some threats have been fixed:
`Not all threats could be fixed. Please contact our support.`